### PR TITLE
docs(readme): adds a readme to the documentation folder

### PR DESCRIPTION
**Pull request by PaulRMellor**
**Documentation**

As we now have an "Edit this page" link from the documentation pages of the Strimzi website that goes to the `/documentation` folder, this readme gives some context around the files and how to make an update . 

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: documentation/README.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+[strimziDoc]: https://strimzi.io/documentation/ "Strimzi documentation"
+[strimziContrib]: https://strimzi.io/contributing/guide/ "Strimzi documentation contributor guide"
+
+<!-- omit from toc -->
+# Strimzi documentation
+
+Welcome to the Strimzi documentation! This repository contains the source files for managing the Strimzi documentation. The documentation is written in AsciiDoc, and it covers guides to help you understand, deploy, and configure Strimzi.
```
Comment (ppatierno) [inline documentation/README.md:7]: I would start "This repository ... " on a new line

File: documentation/README.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  - Files: [/documentation/overview/](overview)
+  - This guide provides an overview of Strimzi, explaining its key concepts and features.
+
+- **Deploying and Managing Strimzi guide**
+  - Files: [/documentation/deploying/](deploying)
+  - This guide provides instructions and best practices for deploying and managing Strimzi.
+
+- **API Reference**
+  - Files: [/documentation/configuring/](configuring)
+  - This guide provides detailed information on configuring Strimzi through the API.
```
Comment (ppatierno) [inline documentation/README.md:29]:
```suggestion
  - This guide provides detailed information on configuring Strimzi through its API.
```

File: documentation/README.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+Other documentation folders contain the content that's incorporated into the main source files.
+
+**Documentation folders**
+| Folder                   | Description                                          |
+| --------------           | -------------------------------------------------    |
+| `api/`                   | Property descriptions for the API Reference guide    |
+| `contributing/`          | Documentation Contributor Guide                      |
+| `deploying/`             | Deploying and Managing Strimzi guide (main)                       |
+| `overview/`              | Strimzi Overview guide (main)                               |
+| `configuring/`           | API Reference (main)                 |
```
Comment (ppatierno) [inline documentation/README.md:49]: the last three rows have the pipe delimiter not aligned with the others.

**Review by ppatierno — APPROVED** [review]
Useful guide! LGTM, I just left some nits (but you can also ignore them if you want)

File: documentation/README.md (hunk 1 above)
Comment (scholzj) [inline documentation/README.md:7]: Should it be `This folder ...`  rather than `This repository ...`? Since it is only one part of it.

**Review by scholzj — COMMENTED** [review]
One nit, LGTM otherwise.

**Review by fvaleri — APPROVED** [review]
This is great @PaulRMellor thanks.

**Review by scholzj — APPROVED** [review]

**Pull request merged**